### ISSUE TITLE
Add support for texture caching using entt

### ISSUE THREAD

**Issue by zachross015** [zachross015/untitled-game-engine#1]
`entt::registry` has methods for managing resources by storing them and retrieving them in the cache as given [here](https://github.com/skypjack/entt/wiki/Crash-Course:-resource-management). 

We want to integrate this with textures as a starting point, since we have yet to deal with any audio. Once this is done, spritesheets should be simple addendum.

**Comment by zachross015** [comment]
Unit testing should be added by, at the very least, storing and loading items from this cache, and storing the item out of scope and checking if it is still in the cache in a different scope.
